src/xml_parser.c

```diff
@@ -292,3 +292,47 @@ cr_xml_parser_generic_from_string(xmlParserCtxtPtr parser,
 
     return ret;
 }
+
+const xmlChar **
+unescape_ampersand_from_values(const xmlChar **attr, gboolean *allocation_needed) {
```

Comment:
It sounds like there's a hard example for `&`, and addressing that is probably enough for now, but is it theoretically possible for other special characters to sneak in?

Reply:
I am not sure what is allowed, [here](https://rpm-software-management.github.io/rpm/manual/spec.html) it says: "Names should not include any numeric operators (‘<’, ‘>’,’=’)" but that's names not capabilities. For capabilities there is just: "Capability names must start with alphanumerics or underscore".

However I tried it with `<` and those don't get encoded like `&` in libxml2. It seems `&` is a special case: https://github.com/GNOME/libxml2/blob/master/parser.c#L3954